users/tests/test_pipeline.py

```diff
@@ -0,0 +1,54 @@
+import pytest
+from django.urls import reverse
+from django.utils.crypto import get_random_string
+from social_core.exceptions import AuthFailed
+
+from tunnistamo.tests.conftest import create_rsa_key, reload_social_django_utils
+from users.tests.conftest import DummyFixedOidcBackend
+
+
+@pytest.fixture
+def dummy_backend(settings):
+    create_rsa_key()
+
+    settings.AUTHENTICATION_BACKENDS = settings.AUTHENTICATION_BACKENDS + (
+        'users.tests.conftest.DummyFixedOidcBackend',
+    )
+    settings.SOCIAL_AUTH_DUMMYFIXEDOIDCBACKEND_OIDC_ENDPOINT = 'https://dummy.example.com'
+    settings.SOCIAL_AUTH_DUMMYFIXEDOIDCBACKEND_KEY = 'tunnistamo'
+    settings.EMAIL_EXEMPT_AUTH_BACKENDS = [DummyFixedOidcBackend.name]
+    settings.ALWAYS_REAUTHENTICATE_BACKENDS = [DummyFixedOidcBackend.name]
```

Comment:
Although `ALWAYS_REAUTHENTICATE_BACKENDS` revealed the bug, it's not the cause of it. This setting doesn't have to be present in these tests. I would remove it, otherwise it could cause confusion why it's here.

Reply:
Ah. True.